File: dev-docs/snippets/rule-sets/delete.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Delete a rule set by its ID. JS returns a typed `{ success: boolean }`.
export async function deleteRuleSet(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const { success } = await client.ruleSets.delete(ruleSetId);
  // #endregion example
  return success;
}

File: dev-docs/snippets/rule-sets/delete.swift
import JsBaoClient

// Delete a rule set by its ID. Swift returns a typed `SuccessResult`, matching
// JS's `{ success: boolean }`.
func deleteRuleSet(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let result = try await client.ruleSets.delete(ruleSetId: ruleSetId)
  let success = result.success
  // #endregion example
  _ = success
}
